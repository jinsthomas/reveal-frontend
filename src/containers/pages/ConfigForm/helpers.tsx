import React from 'react';
import Loading from '../../../components/page/Loading';
import { OPENSRP_API_BASE_URL } from '../../../configs/env';
import { growl } from '../../../helpers/utils';
import { getPayloadOptions } from '../../../services/opensrp';

export const defaultConfigProps = {
  LoadingComponent: <Loading />,
  baseURL: OPENSRP_API_BASE_URL,
  getPayload: getPayloadOptions,
  growl,
};
