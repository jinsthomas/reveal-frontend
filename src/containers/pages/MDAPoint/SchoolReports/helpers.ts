import { Page } from '../../../../components/page/HeaderBreadcrumb/HeaderBreadcrumb';
import { MDA_POINT_SCHOOL_REPORT_TITLE } from '../../../../configs/lang';
import { SchoolReport } from '../../../../store/ducks/generic/MDASchoolReport';

interface ReturnPagedata {
  pageTitle: string;
  prevPage: Page;
}

export const getPrevPageAndTitle = (school: SchoolReport): ReturnPagedata => {
  const depth = school.jurisdiction_depth;
  const prevPage = {
    label: school.jurisdiction_name_path[depth - 1],
    url: school.jurisdiction_id_path[depth - 1],
  };
  const pageTitle = `${MDA_POINT_SCHOOL_REPORT_TITLE}: ${school.jurisdiction_name_path[depth]}`;
  return {
    pageTitle,
    prevPage,
  };
};
