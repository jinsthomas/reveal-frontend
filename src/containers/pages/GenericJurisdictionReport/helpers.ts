import { Cell, Column } from 'react-table';
<<<<<<< HEAD:src/containers/pages/GenericJurisdictionReport/helpers.ts
import { getIRSThresholdAdherenceIndicator } from '../../../helpers/indicators';
=======
import { getIRSThresholdAdherenceIndicator } from '../../../../helpers/indicators';
>>>>>>> Replacing typings and props dependent on react-table-v6:src/containers/pages/IRS/JurisdictionsReport/helpers.ts

/** columns for Namibia IRS jurisdictions */
export const NamibiaColumns = [
  {
    Header: 'Name',
    columns: [
      {
        accessor: 'jurisdiction_name',
        minWidth: 180,
      },
    ],
  },
  {
    Header: 'Structures Targeted',
    columns: [
      {
        accessor: 'target_2019',
      },
    ],
  },
  {
    Header: 'Structures Found',
    columns: [
      {
        accessor: 'structuresfound',
      },
    ],
  },
  {
    Header: 'Structures Sprayed',
    columns: [
      {
        accessor: 'structuressprayed',
      },
    ],
  },
  {
    Header: 'Target Coverage',
    columns: [
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell),

        accessor: 'targetcoverage',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Found Coverage',
    columns: [
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell),

        accessor: 'foundcoverage',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Refusals',
    columns: [
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell, null),
        Header: 'Following first visit',
        accessor: 'refusalsfirst',
        className: 'indicator centered',
        id: 'refusalsfirst',
      },
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell, null),
        Header: 'Following mop-up',
        accessor: 'refusalsmopup',
        className: 'indicator centered',
        id: 'refusalsmopup',
      },
    ],
    headerClassName: 'centered',
  },
  {
    Header: 'Locked',
    columns: [
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell, null),
        Header: 'Following first visit',
        accessor: 'lockedfirst',
        className: 'indicator centered',
        id: 'lockedfirst',
      },
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell, null),
        Header: 'Following mop-up',
        accessor: 'lockedmopup',
        className: 'indicator centered',
        id: 'lockedmopup',
      },
    ],
    headerClassName: 'centered',
  },
];

/** columns for Zambia IRS jurisdictions */
export const ZambiaJurisdictionsColumns = [
  {
    Header: 'Name',
    columns: [
      {
        Header: '',
        accessor: 'jurisdiction_name',
        minWidth: 180,
      },
    ],
  },
  {
    Header: 'Total Spray Areas',
    columns: [
      {
        Header: '',
        accessor: 'totareas',
      },
    ],
  },
  {
    Header: 'Targeted Spray Areas',
    columns: [
      {
        Header: '',
        accessor: 'targareas',
      },
    ],
  },
  {
    Header: '% visited Spray Areas Effectively sprayed',
    columns: [
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'perctvisareaseffect',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Total Structures',
    columns: [
      {
        Header: '',
        accessor: 'totstruct',
      },
    ],
  },
  {
    Header: 'Targeted Structures',
    columns: [
      {
        Header: '',
        accessor: 'targstruct',
      },
    ],
  },
  {
    Header: 'Sprayed Structures',
    columns: [
      {
        Header: '',
        accessor: 'sprayedstruct',
      },
    ],
  },
  {
    Header: 'Spray coverage of targeted (Progress)',
    columns: [
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraycovtarg',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Room coverage of structures sprayed',
    columns: [
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'roomcov',
        className: 'indicator centered',
      },
    ],
  },
];

/** columns for  mda point jurisdictions */
export const mdaJurisdictionsColumns = [
  {
    Header: 'Name',
    columns: [
      {
        Header: '',
        accessor: 'jurisdiction_name',
        minWidth: 180,
      },
    ],
  },
  {
    Header: 'Total SACs Registered',
    columns: [
      {
        Header: '',
        accessor: 'sacregistered',
      },
    ],
  },
  {
    Header: 'MMA Coverage',
    columns: [
      {
        Header: '',
        accessor: 'mmacov',
      },
    ],
  },
  {
    Header: 'MMA Coverage (%)',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'mmacovper',
      },
    ],
  },
  {
    Header: 'SACs Refused',
    columns: [
      {
        Header: '',
        accessor: 'sacrefused',
      },
    ],
  },
  {
    Header: 'SACs Sick/Pregnant/Contraindicated',
    columns: [
      {
        Header: '',
        accessor: 'sacrefmedreason',
      },
    ],
  },
  {
    Header: 'ADR Reported (%)',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'mmaadr',
      },
    ],
  },
  {
    Header: 'ADR Severe (%)',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'mmaadrsev',
      },
    ],
  },
  {
    Header: 'Alb Tablets Distributed',
    columns: [
      {
        Header: '',
        accessor: 'albdist',
      },
    ],
  },
];

/** columns for Namibia IRS focus (spray) areas */
export const ZambiaFocusAreasColumns = [
  {
    Header: 'Name',
    columns: [
      {
        Header: '',
        accessor: 'jurisdiction_name',
        minWidth: 180,
      },
    ],
  },
  {
    Header: 'Structures on the ground',
    columns: [
      {
        Header: '',
        accessor: 'totstruct',
      },
    ],
  },
  {
    Header: 'Found',
    columns: [
      {
        Header: '',
        accessor: 'foundstruct',
      },
    ],
  },
  {
    Header: 'Sprayed',
    columns: [
      {
        Header: '',
        accessor: 'sprayedstruct',
      },
    ],
  },
  {
    Header: 'Spray Coverage (Effectiveness)',
    columns: [
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraycov',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Found Coverage',
    columns: [
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraytarg',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Spray Success Rate (PMI SC)',
    columns: [
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraysuccess',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Room coverage of structures sprayed',
    columns: [
      {
        Cell: (cell: Cell) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'roomcov',
        className: 'indicator centered',
      },
    ],
  },
];

/** IRS Table Columns
 * These are all the table columns for IRS that we know about.
 */
export const plansTableColumns: { [key: string]: Column[] } = {
  mdaJurisdictionsColumns,
  namibia2019: NamibiaColumns,
  zambiaFocusArea2019: ZambiaFocusAreasColumns,
  zambiaJurisdictions2019: ZambiaJurisdictionsColumns,
};
