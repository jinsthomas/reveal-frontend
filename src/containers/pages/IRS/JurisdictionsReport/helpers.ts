import { CellInfo } from 'react-table';
import { getIRSThresholdAdherenceIndicator } from '../../../../helpers/indicators';
import { FlexObject } from '../../../../helpers/utils';

/** columns for Namibia IRS jurisdictions */
export const NamibiaColumns = [
  {
    Header: 'Name',
    columns: [
      {
        Header: '',
        accessor: 'jurisdiction_name',
        minWidth: 180,
      },
    ],
  },
  {
    Header: 'Structures Targeted',
    columns: [
      {
        Header: '',
        accessor: 'target_2019',
      },
    ],
  },
  {
    Header: 'Structures Found',
    columns: [
      {
        Header: '',
        accessor: 'structuresfound',
      },
    ],
  },
  {
    Header: 'Structures Sprayed',
    columns: [
      {
        Header: '',
        accessor: 'structuressprayed',
      },
    ],
  },
  {
    Header: 'Target Coverage',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'targetcoverage',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Found Coverage',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'foundcoverage',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Refusals',
    columns: [
      {
        Header: 'Following first visit',
        accessor: 'refusalsfirst',
        className: 'centered',
        id: 'refusalsfirst',
      },
      {
        Header: 'Following mop-up',
        accessor: 'refusalsmopup',
        className: 'centered',
        id: 'refusalsmopup',
      },
    ],
  },
  {
    Header: 'Locked',
    columns: [
      {
        Header: 'Following first visit',
        accessor: 'lockedfirst',
        className: 'centered',
        id: 'lockedfirst',
      },
      {
        Header: 'Following mop-up',
        accessor: 'lockedmopup',
        className: 'centered',
        id: 'lockedmopup',
      },
    ],
  },
];

/** columns for Zambia IRS jurisdictions */
export const ZambiaJurisdictionsColumns = [
  {
    Header: 'Name',
    columns: [
      {
        Header: '',
        accessor: 'jurisdiction_name',
        minWidth: 180,
      },
    ],
  },
  {
    Header: 'Total Spray Areas',
    columns: [
      {
        Header: '',
        accessor: 'totareas',
      },
    ],
  },
  {
    Header: 'Targeted Spray Areas',
    columns: [
      {
        Header: '',
        accessor: 'targareas',
      },
    ],
  },
  {
    Header: '% visited Spray Areas Effectively sprayed',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'perctvisareaseffect',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Total Structures',
    columns: [
      {
        Header: '',
        accessor: 'totstruct',
      },
    ],
  },
  {
    Header: 'Targeted Structures',
    columns: [
      {
        Header: '',
        accessor: 'targstruct',
      },
    ],
  },
  {
    Header: 'Sprayed Structures',
    columns: [
      {
        Header: '',
        accessor: 'sprayedstruct',
      },
    ],
  },
  {
    Header: 'Spray coverage of targeted (Progress)',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraycovtarg',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Room coverage of structures sprayed',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'roomcov',
        className: 'indicator centered',
      },
    ],
  },
];

/** columns for Namibia IRS focus (spray) areas */
export const ZambiaFocusAreasColumns = [
  {
    Header: 'Name',
    columns: [
      {
        Header: '',
        accessor: 'jurisdiction_name',
        minWidth: 180,
      },
    ],
  },
  {
    Header: 'Structures on the ground',
    columns: [
      {
        Header: '',
        accessor: 'totstruct',
      },
    ],
  },
  {
    Header: 'Found',
    columns: [
      {
        Header: '',
        accessor: 'foundstruct',
      },
    ],
  },
  {
    Header: 'Sprayed',
    columns: [
      {
        Header: '',
        accessor: 'sprayedstruct',
      },
    ],
  },
  {
    Header: 'Spray Coverage (Effectiveness)',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraycov',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Found Coverage',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraytarg',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Spray Success Rate (PMI SC)',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'spraysuccess',
        className: 'indicator centered',
      },
    ],
  },
  {
    Header: 'Room coverage of structures sprayed',
    columns: [
      {
        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
        Header: '',
        accessor: 'roomcov',
        className: 'indicator centered',
      },
    ],
  },
];

/** extract hierarchical tree of jurisdictions as a list, with the highest
 * level item in the hierarchy as the 0th element
 * @param {FlexObject[]} tree - the data set
 * @param {FlexObject} item - the item in question
 * @returns {FlexObject[]} - the hierarchy for the item in question
 */
export function getTree(
  tree: FlexObject[],
  item: FlexObject,
  result: FlexObject[] = []
): FlexObject[] {
  if (item.jurisdiction_parent_id === null && !result.includes(item)) {
    result.push(item);
  } else {
    const parentItems = tree.filter(el => el.jurisdiction_id === item.jurisdiction_parent_id);
    if (parentItems.length > 0) {
      if (!result.includes(parentItems[0])) {
        result.push(parentItems[0]);
      }
      return getTree(tree, parentItems[0], result);
    }
  }
  return result;
}
