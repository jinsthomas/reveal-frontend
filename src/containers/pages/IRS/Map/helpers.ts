import GeojsonExtent from '@mapbox/geojson-extent';
import { BLACK, GREY, TASK_GREEN, TASK_ORANGE, TASK_RED, TASK_YELLOW } from '../../../../colors';
import { GisidaProps } from '../../../../components/GisidaWrapper';
import { circleLayerConfig, fillLayerConfig, lineLayerConfig } from '../../../../configs/settings';
import { MAIN_PLAN, STRUCTURE_LAYER } from '../../../../constants';
import { FlexObject } from '../../../../helpers/utils';
import { IRSJurisdiction } from '../../../../store/ducks/IRS/jurisdictions';
import { StructureFeatureCollection } from '../../../../store/ducks/IRS/structures';
import { Jurisdiction } from '../../../../store/ducks/jurisdictions';

/** Default indicator stops */
const defaultIndicatorStops = [
  ['Complete', TASK_GREEN],
  ['Not Sprayed', TASK_RED],
  ['Partially Sprayed', TASK_ORANGE],
  ['Not Visited', TASK_YELLOW],
  ['Not Eligible', BLACK],
];

/** Utility function to build structure layer
 * @param {StructureFeatureCollection} structures - Feature Collection of structures
 * @param {string[][]} indicatorStops - the indicator stops
 */
export const structuresLayerBuilder = (
  structures: StructureFeatureCollection,
  indicatorStops: string[][] = defaultIndicatorStops
) => {
  const structuresLayers: FlexObject[] = [];
  const layerType = structures.features[0].geometry && structures.features[0].geometry.type;
  const structuresPopup: FlexObject = {
    body: `<div>
          <p class="heading">{{structure_type}}</p>
          <p>Status: {{business_status}}</p>
        </div>`,
    join: ['structure_jurisdiction_id', 'structure_jurisdiction_id'],
  };

  const structureStatusColors = {
    default: GREY,
    property: 'business_status',
    stops: indicatorStops,
    type: 'categorical',
  };

  if (layerType === 'Point') {
    // build circle layers if structures are points
    const structureCircleLayer = {
      ...circleLayerConfig,
      id: `${STRUCTURE_LAYER}-circle`,
      paint: {
        ...circleLayerConfig.paint,
        'circle-color': structureStatusColors,
        'circle-radius': ['interpolate', ['linear'], ['zoom'], 13, 2, 14, 4, 16, 8, 20, 12],
        'circle-stroke-color': structureStatusColors,
        'circle-stroke-opacity': 1,
      },
      popup: structuresPopup,
      source: {
        ...circleLayerConfig.source,
        data: {
          data: JSON.stringify(structures),
          type: 'stringified-geojson',
        },
        type: 'geojson',
      },
      visible: true,
    };
    structuresLayers.push(structureCircleLayer);
  } else {
    // build fill / line layers if structures are polygons
    const structuresFillLayer = {
      ...fillLayerConfig,
      id: `${STRUCTURE_LAYER}-fill`,
      paint: {
        ...fillLayerConfig.paint,
        'fill-color': structureStatusColors,
        'fill-outline-color': structureStatusColors,
      },
      popup: structuresPopup,
      source: {
        ...fillLayerConfig.source,
        data: {
          ...fillLayerConfig.source.data,
          data: JSON.stringify(structures),
        },
      },
      visible: true,
    };
    structuresLayers.push(structuresFillLayer);

    const structuresLineLayer = {
      ...lineLayerConfig,
      id: `${STRUCTURE_LAYER}-line`,
      paint: {
        'line-color': structureStatusColors,
        'line-opacity': 1,
        'line-width': 2,
      },
      source: {
        ...lineLayerConfig.source,
        data: {
          ...lineLayerConfig.source.data,
          data: JSON.stringify(structures),
        },
      },
    };
    structuresLayers.push(structuresLineLayer);
  }
  return structuresLayers;
};

/** Get Gisida Wrapper Props
 * @param {Jurisdiction} jurisdiction - the jurisdiction (with geojson field)
 * @param {StructureFeatureCollection} structures - Feature Collection of structures
 * @param {string[][]} indicatorStops - the indicator stops
 */
export const getGisidaWrapperProps = (
  jurisdiction: Jurisdiction,
  structures: StructureFeatureCollection | null,
  indicatorStops: string[][] = defaultIndicatorStops
): GisidaProps | null => {
  if (!jurisdiction.geojson) {
    return null;
  }
  const layers: FlexObject[] = [];

  // define line layer for Jurisdiction outline
  const jurisdictionLineLayer = {
    ...lineLayerConfig,
    id: `${MAIN_PLAN}-${jurisdiction.jurisdiction_id}`,
    source: {
      ...lineLayerConfig.source,
      data: {
        ...lineLayerConfig.source.data,
        data: JSON.stringify(jurisdiction.geojson),
      },
    },
    visible: true,
  };
  layers.push(jurisdictionLineLayer);

  // Define structures layers
  let structuresLayers: FlexObject[] = [];
  if (structures) {
    structuresLayers = structuresLayerBuilder(structures, indicatorStops);
  }

  // define feature collection of all geoms being rendered
  const featureCollection =
    !structures || !structures.features || !structures.features.length
      ? { features: [jurisdiction.geojson], type: 'FeatureCollection' }
      : {
          ...structures,
          features: [...structures.features, jurisdiction.geojson],
        };
  // define bounds for gisida map position
  const bounds = GeojsonExtent(featureCollection);

  for (const structureLayer of structuresLayers) {
    layers.push(structureLayer);
  }

  const gisidaWrapperProps: GisidaProps = {
    bounds,
    geoData: null,
    handlers: [],
    layers,
    pointFeatureCollection: null,
    polygonFeatureCollection: null,
    structures: null,
  };
  return gisidaWrapperProps;
};

/** Get breadcrumbs for a jurisdiction object
 * This uses the jurisdiction_name_path and jurisdiction_path fields to get
 * the breadcrumbs of the parents of a given jurisdiction object
 * @param {IRSJurisdiction} jurisdiction - the jurisdiction in question
 * @param {string} urlPath - the url path that we append the jurisdiction id to
 */
export const getJurisdictionBreadcrumbs = (
  jurisdiction: IRSJurisdiction,
  urlPath: string = '/'
) => {
  const result = [];

  for (let i = 0; i < jurisdiction.jurisdiction_name_path.length; i++) {
    let url = urlPath;
    if (jurisdiction.jurisdiction_path[i]) {
      url = `${urlPath}/${jurisdiction.jurisdiction_path[i]}`;
    }
    result.push({
      label: jurisdiction.jurisdiction_name_path[i],
      url,
    });
  }

  return result;
};
