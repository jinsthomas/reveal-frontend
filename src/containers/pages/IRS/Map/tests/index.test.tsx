import reducerRegistry from '@onaio/redux-reducer-registry';
import superset from '@onaio/superset-connector';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import flushPromises from 'flush-promises';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Helmet } from 'react-helmet';
import { Router } from 'react-router';
import { IRSReportingMap } from '../';
import { INTERVENTION_IRS_URL, IRS_REPORTING_TITLE, MAP } from '../../../../../constants';
import store from '../../../../../store';
import GenericJurisdictionsReducer, {
  fetchGenericJurisdictions,
  GenericJurisdiction,
  getGenericJurisdictionByJurisdictionId,
  reducerName as GenericJurisdictionsReducerName,
} from '../../../../../store/ducks/generic/jurisdictions';
import IRSPlansReducer, {
  fetchIRSPlans,
  getIRSPlanById,
  IRSPlan,
  reducerName as IRSPlansReducerName,
} from '../../../../../store/ducks/generic/plans';
import genericStructuresReducer, {
  fetchGenericStructures,
  GenericStructure,
  getGenericStructures,
  reducerName as genericStructuresReducerName,
  StructureFeatureCollection,
} from '../../../../../store/ducks/generic/structures';
import { plans } from '../../../../../store/ducks/generic/tests/fixtures';
import jurisdictionReducer, {
  fetchJurisdictions,
  getJurisdictionById,
  Jurisdiction,
  reducerName as jurisdictionReducerName,
} from '../../../../../store/ducks/jurisdictions';
import * as fixtures from '../../JurisdictionsReport/tests/fixtures';
import { getGisidaWrapperProps } from '../helpers';

/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

/** register the reducers */
reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);
reducerRegistry.register(jurisdictionReducerName, jurisdictionReducer);
reducerRegistry.register(GenericJurisdictionsReducerName, GenericJurisdictionsReducer);
reducerRegistry.register(genericStructuresReducerName, genericStructuresReducer);

/** set up data in the store needed for this view */
const focusAreaData = superset.processData(fixtures.ZambiaFocusAreasJSON) || [];
const structureData = superset.processData(fixtures.ZambiaStructuresJSON) || [];
const jurisdictionData = superset.processData(fixtures.ZambiaAkros1JSON) || [];

store.dispatch(fetchGenericJurisdictions('zm-focusAreas', focusAreaData));
store.dispatch(fetchGenericStructures('zm-structures', structureData));
store.dispatch(fetchJurisdictions(jurisdictionData));

const history = createBrowserHistory();

describe('components/IRS Reports/IRSReportingMap', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const mock: any = jest.fn();
    const props = {
      focusArea: getGenericJurisdictionByJurisdictionId(
        store.getState(),
        'zm-focusAreas',
        '0dc2d15b-be1d-45d3-93d8-043a3a916f30'
      ),
      history,
      jurisdiction: getJurisdictionById(store.getState(), '0dc2d15b-be1d-45d3-93d8-043a3a916f30'),
      location: mock,
      match: {
        isExact: true,
        params: {
          jurisdictionId: '0dc2d15b-be1d-45d3-93d8-043a3a916f30',
          planId: (plans[0] as IRSPlan).plan_id,
        },
        path: `${INTERVENTION_IRS_URL}/:planId/:jurisdictionId/${MAP}`,
        url: `${INTERVENTION_IRS_URL}/${
          (plans[0] as IRSPlan).plan_id
        }/0dc2d15b-be1d-45d3-93d8-043a3a916f30/${MAP}`,
      },
      plan: plans[0] as IRSPlan,
      structures: getGenericStructures(
        store.getState(),
        'zm-structures',
        '0dc2d15b-be1d-45d3-93d8-043a3a916f30'
      ),
    };
    shallow(
      <Router history={history}>
        <IRSReportingMap {...props} />
      </Router>
    );
  });

  it('renders IRSReportingMap correctly', async () => {
    fetch.mockResponseOnce(JSON.stringify({}));
    const mock: any = jest.fn();

    const focusArea = getGenericJurisdictionByJurisdictionId(
      store.getState(),
      'zm-focusAreas',
      '0dc2d15b-be1d-45d3-93d8-043a3a916f30'
    );

    const structures = getGenericStructures(
      store.getState(),
      'zm-structures',
      '0dc2d15b-be1d-45d3-93d8-043a3a916f30'
    );

    const jurisdiction = getJurisdictionById(
      store.getState(),
      '0dc2d15b-be1d-45d3-93d8-043a3a916f30'
    );

    const props = {
      focusArea,
      history,
      jurisdiction,
      location: mock,
      match: {
        isExact: true,
        params: {
          jurisdictionId: '0dc2d15b-be1d-45d3-93d8-043a3a916f30',
          planId: (plans[0] as IRSPlan).plan_id,
        },
        path: `${INTERVENTION_IRS_URL}/:planId/:jurisdictionId/${MAP}`,
        url: `${INTERVENTION_IRS_URL}/${
          (plans[0] as IRSPlan).plan_id
        }/0dc2d15b-be1d-45d3-93d8-043a3a916f30/${MAP}`,
      },
      plan: plans[0] as IRSPlan,
      structures,
    };
    const wrapper = mount(
      <Router history={history}>
        <IRSReportingMap {...props} />
      </Router>
    );
    const helmet = Helmet.peek();
    await flushPromises();
    expect(toJson(wrapper.find('BreadcrumbItem li'))).toMatchSnapshot('breadcrumbs');
    expect(toJson(wrapper.find('h3.page-title'))).toMatchSnapshot('page title');
    expect(helmet.title).toEqual('IRS 2019-09-05 TEST');
    expect(toJson(wrapper.find('mapSidebar h5'))).toMatchSnapshot('Sidebar title');
    expect(toJson(wrapper.find('sidebar-legend-item'))).toMatchSnapshot('Legend items');
    expect(toJson(wrapper.find('responseItem'))).toMatchSnapshot('Response items');
    expect(wrapper.find('GisidaWrapper').props()).toEqual(
      getGisidaWrapperProps(jurisdiction as Jurisdiction, structures)
    );
    wrapper.unmount();
  });
});
