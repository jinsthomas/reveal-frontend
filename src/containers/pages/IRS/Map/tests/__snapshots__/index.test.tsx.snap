// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IRS Reports/IRSReportingMap renders IRSReportingMap correctly: Legend items 1`] = `null`;

exports[`components/IRS Reports/IRSReportingMap renders IRSReportingMap correctly: Response items 1`] = `null`;

exports[`components/IRS Reports/IRSReportingMap renders IRSReportingMap correctly: Sidebar title 1`] = `null`;

exports[`components/IRS Reports/IRSReportingMap renders IRSReportingMap correctly: breadcrumbs 1`] = `
Array [
  <li
    className="breadcrumb-item"
  >
    <Link
      key="0"
      replace={false}
      to="/"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="1"
      replace={false}
      to="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce"
    >
      <a
        href="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce"
        onClick={[Function]}
      >
        IRS Reporting
      </a>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="2"
      replace={false}
      to="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce/22bc44dd-752d-4c20-8761-617361b4f1e7"
    >
      <a
        href="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce/22bc44dd-752d-4c20-8761-617361b4f1e7"
        onClick={[Function]}
      >
        Zambia
      </a>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="3"
      replace={false}
      to="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce/fd5a22ba-fa0e-4fdc-ae1c-1db8965b7b43"
    >
      <a
        href="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce/fd5a22ba-fa0e-4fdc-ae1c-1db8965b7b43"
        onClick={[Function]}
      >
        Lusaka
      </a>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="4"
      replace={false}
      to="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce/8f1fd4cf-264d-4ed6-8838-e538c5845d46"
    >
      <a
        href="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce/8f1fd4cf-264d-4ed6-8838-e538c5845d46"
        onClick={[Function]}
      >
        Lusaka_Test
      </a>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="5"
      replace={false}
      to="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce/2bf9915d-8725-4061-983d-5938802ac0f0"
    >
      <a
        href="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce/2bf9915d-8725-4061-983d-5938802ac0f0"
        onClick={[Function]}
      >
        Lusaka HFC
      </a>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="6"
      replace={false}
      to="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce/0dc2d15b-be1d-45d3-93d8-043a3a916f30"
    >
      <a
        href="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce/0dc2d15b-be1d-45d3-93d8-043a3a916f30"
        onClick={[Function]}
      >
        Akros_1
      </a>
    </Link>
  </li>,
  <li
    aria-current="page"
    className="active breadcrumb-item"
  >
    Akros_1 map
  </li>,
]
`;

exports[`components/IRS Reports/IRSReportingMap renders IRSReportingMap correctly: page title 1`] = `
<h3
  className="mb-3 page-title"
>
  IRS 2019-09-05 TEST
</h3>
`;
